src/types/widget.ts

```diff
@@ -115,7 +115,31 @@ export type UiState = {
  * have at least a `render` or a `init` function.
  */
 export interface Widget {
-  $$type?: string;
+  $$type?:
```

Comment:
The type is set at the connector level not at the widget level. Currently this type is incorrect.

Reply:
Would you see this type in `WidgetFactory` instead?

Reply:
Yes it could be.

Reply:
Let's change that in another PR if you don't mind since it's not related to this change and could lead to broader TypeScript changes.